### ISSUE TITLE
Please add extended support for SF Symbol configuration

### ISSUE THREAD

**Issue by funnel20** [futuretap/InAppSettingsKit#489]
I'm very pleased that InAppSettingsKit version 3.6 now supports SF Symbols via `systemImageNamed:`.

One question though, as this opens the door to more customisation. In our apps we use even more native power with [`systemImageNamed:withConfiguration:`](https://developer.apple.com/documentation/uikit/uiimage/3294234-systemimagenamed).
For example to apply palette colouring on a SF Symbol:
<img width="46" alt="Screenshot 2024-02-16 at 10 23 29" src="https://github.com/futuretap/InAppSettingsKit/assets/2139844/67058db2-53dc-4b03-aab5-5e84bf9252ad">

```ObjC
// Define image name and configuration:
NSString *name = @"cloud.sun";
UIImageConfiguration *config = [UIImageSymbolConfiguration configurationWithPaletteColors:@[
    [UIColor systemTealColor],      // cloud
    [UIColor systemYellowColor]     // sun
]];

// Create image:
UIImage *image = [UIImage systemImageNamed:name
                         withConfiguration:config];
```
Since you already use `IASKCellImage` as `name`, would it be possible to extend InAppSettingsKit with something like a `IASKCellImageConfiguration`, which refers to the name of Class variable of the parent ViewController (e.g. `config` in this example)?

The good part of `systemImageNamed:withConfiguration:` is that even if `config` is `nil`, it will ignore the configuration and just acts as `systemImageNamed:`.

Thank you for considering this improvement.

**Comment by futuretap** [comment]
This is certainly a good idea. I suggest adding a `symbolImageConfiguration` property to `IASKAppSettingsViewController`. That property would have to be copied to all child view controllers (similar to how the `delegate` is assigned to child VCs). Is probably easier to use, especially from Swift code.

Would you be so kind to contribute a pull request? ;-)

**Comment by funnel20** [comment]
@futuretap I was hoping to apply multiple **different** Symbol Configurations per ViewController, as the layout might request one configuration per image. 
Consider for example the native iOS Settings app, it consists of a wide range of coloured icons:

![IMG_8B84F6225CA7-1](https://github.com/futuretap/InAppSettingsKit/assets/2139844/8e0d734d-4a44-45ea-aac5-7dc198c4e830)

Hence my idea of a new parameter in the `.plist`

**Comment by futuretap** [comment]
I see. To achieve this example, you’d need a separate class for each background color. Doesn't really scale well, too.
If I were you, I'd implement the `-settingsViewController:cellForSpecifier:` delegate callback and set 

`cell.imageView.image = cell.imageView.image.withConfiguration(yourConfiguration)`

Then you can switch on `specifier.key` to create a configuration per key. Does that work for you?

**Comment by funnel20** [comment]
@futuretap That's a great idea, thank you very much.

**Comment by funnel20** [comment]
> I see. To achieve this example, you’d need a separate class for each background color. Doesn't really scale well, too. If I were you, I'd implement the `-settingsViewController:cellForSpecifier:` delegate callback and set
> 
> `cell.imageView.image = cell.imageView.image.withConfiguration(yourConfiguration)`
> 
> Then you can switch on `specifier.key` to create a configuration per key. Does that work for you?

@futuretap I have tried this via:
```ObjC
- (UITableViewCell*)settingsViewController:(UITableViewController<IASKViewController>*)settingsViewController
                          cellForSpecifier:(IASKSpecifier*)specifier {
    NSLog(@"settingsViewController:cellForSpecifier:");
    
    NSIndexPath* indexPath = [settingsViewController.settingsReader indexPathForKey:specifier.key];
    UITableViewCell* cell = [self.tableView cellForRowAtIndexPath:indexPath];
    
    if ([specifier.key isEqualToString:@"TEST"]) {
        NSString *name = @"cloud.sun";
        UIImageConfiguration *config = [UIImageSymbolConfiguration configurationWithPaletteColors:@[
            [UIColor systemTealColor],      // cloud
            [UIColor systemYellowColor]     // sun
        ]];

        // Create image:
        UIImage *image = [UIImage systemImageNamed:name
                                 withConfiguration:config];
        
        cell.imageView.image = image;
    }
    
    return cell;
}
```

But the delegate method is never called (I don't see the `NSLog()`). It will be only called for a Custom View.
But I'd like to apply the SF Configuration on a standard cell, for example a `PSChildPaneSpecifier`.

Would it be an idea to add a new delegate to customize the image based on identifier? E.g. like:
```ObjC
- (UIImage*)settingsViewController:(UITableViewController<IASKViewController>*)settingsViewController
                 imageForSpecifier:(IASKSpecifier*)specifier;
```

**Comment by futuretap** [comment]
Indeed, the delegate is only called for `IASKCustomViewSpecifier` cells (as documented).
Alternatively, you can subclass `IASKAppSettingsViewController` and override `-tableView:cellForRowAtIndexPath:`. 

There are so many different properties of a cell that imo it doesn’t scale to add delegate callbacks for all of these. We might rather think of a `customizeCell` callback that allows the delegate to manipulate any cell after IASK has done its work.
